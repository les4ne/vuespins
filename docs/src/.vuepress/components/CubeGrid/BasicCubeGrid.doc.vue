<template>
  <div>
    <CubeGrid />
  </div>
</template>

<script>
import { CubeGrid } from 'vuespins'

export default {
  name: 'CubeGrid',
  components: { CubeGrid },
}
</script>
