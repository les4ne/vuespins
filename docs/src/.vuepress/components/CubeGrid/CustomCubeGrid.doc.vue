<template>
  <div>
    <CubeGrid color="#fb503b" :size="80" :speed="5" />
  </div>
</template>

<script>
import { CubeGrid } from 'vuespins'

export default {
  name: 'app',
  components: { CubeGrid }
}
</script>
