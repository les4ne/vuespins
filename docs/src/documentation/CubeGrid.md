# CubeGrid

## **Basic**

<br>
<client-only>
<CubeGrid />
</client-only>

### Props

Basic "CubeGrid" component

| property | propType | required | default | custom |
| :------: | :------: | :------: | :-----: | :----: |
| `color`  |  string  |  false   | #282c34 |   -    |
|  `size`  |  string  |  false   |   50    |   -    |
| `speed`  |  string  |  false   |   1.3   |   -    |

### Usage

<<< @/src/.vuepress/components/CubeGrid/BasicCubeGrid.doc.vue{3,8,10-13}

## **Custom**

<br>
<client-only>
<CubeGrid color="#fb503b" size="80" speed="5" />
</client-only>

### Props

Custom "CubeGrid" component

| property | propType | required | default | custom  |
| :------: | :------: | :------: | :-----: | :-----: |
| `color`  |  string  |  false   | #282c34 | #fb503b |
|  `size`  |  string  |  false   |   50    |   80    |
| `speed`  |  string  |  false   |   1.3   |    5    |

### Usage

<<< @/src/.vuepress/components/CubeGrid/CustomCubeGrid.doc.vue{3,8,10-13}
