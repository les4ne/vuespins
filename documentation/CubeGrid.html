<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>CubeGrid | vuespins</title>
    <meta name="generator" content="VuePress 1.5.4">
    
    <meta name="description" content="Css Spinners for Vue">
    <meta name="theme-color" content="#3eaf7c">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="preload" href="/assets/css/0.styles.c37a39fb.css" as="style"><link rel="preload" href="/assets/js/app.2a57d547.js" as="script"><link rel="preload" href="/assets/js/2.d688e986.js" as="script"><link rel="preload" href="/assets/js/11.03618cb3.js" as="script"><link rel="prefetch" href="/assets/js/10.aa959343.js"><link rel="prefetch" href="/assets/js/12.1f507013.js"><link rel="prefetch" href="/assets/js/13.506649eb.js"><link rel="prefetch" href="/assets/js/14.b1e1242f.js"><link rel="prefetch" href="/assets/js/15.a4c33060.js"><link rel="prefetch" href="/assets/js/16.3da839f4.js"><link rel="prefetch" href="/assets/js/3.fa382cb2.js"><link rel="prefetch" href="/assets/js/4.85f34f4c.js"><link rel="prefetch" href="/assets/js/5.bdf32507.js"><link rel="prefetch" href="/assets/js/6.10c6ee39.js"><link rel="prefetch" href="/assets/js/7.4de9ee27.js"><link rel="prefetch" href="/assets/js/8.f8ce8af7.js"><link rel="prefetch" href="/assets/js/9.544bcd7a.js">
    <link rel="stylesheet" href="/assets/css/0.styles.c37a39fb.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><img src="https://raw.githubusercontent.com/vuejs/art/master/logo-monochrome.svg" alt="vuespins" class="logo"> <span class="site-name can-hide">vuespins</span></a> <div class="links"><!----> <nav class="nav-links can-hide"><div class="nav-item"><a href="/documentation/installation.html" class="nav-link">
  Docs
</a></div> <a href="https://pwndex.github.io/vuespins/" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/documentation/installation.html" class="nav-link">
  Docs
</a></div> <a href="https://pwndex.github.io/vuespins/" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>Documentation</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/documentation/Installation.html" class="sidebar-link">Installation</a></li><li><a href="/documentation/Ring.html" class="sidebar-link">Ring</a></li><li><a href="/documentation/CubeGrid.html" aria-current="page" class="active sidebar-link">CubeGrid</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/documentation/CubeGrid.html#basic" class="sidebar-link">Basic</a></li><li class="sidebar-sub-header"><a href="/documentation/CubeGrid.html#custom" class="sidebar-link">Custom</a></li></ul></li><li><a href="/documentation/Spinner.html" class="sidebar-link">Spinner</a></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="cubegrid"><a href="#cubegrid" class="header-anchor">#</a> CubeGrid</h1> <h2 id="basic"><a href="#basic" class="header-anchor">#</a> <strong>Basic</strong></h2> <br> <!----> <h3 id="props"><a href="#props" class="header-anchor">#</a> Props</h3> <p>Basic &quot;CubeGrid&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">-</td></tr></tbody></table> <h3 id="usage"><a href="#usage" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>CubeGrid</span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><h2 id="custom"><a href="#custom" class="header-anchor">#</a> <strong>Custom</strong></h2> <br> <!----> <h3 id="props-2"><a href="#props-2" class="header-anchor">#</a> Props</h3> <p>Custom &quot;CubeGrid&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">#fb503b</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">80</td></tr></tbody></table> <h3 id="usage-2"><a href="#usage-2" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>CubeGrid</span> <span class="token attr-name">color</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>#fb503b<span class="token punctuation">&quot;</span></span> <span class="token attr-name">size</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>80<span class="token punctuation">&quot;</span></span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div> <footer class="page-edit"><div class="edit-link"><a href="https://pwndex.github.io/vuespins/edit/master/docs/src/documentation/CubeGrid.md" target="_blank" rel="noopener noreferrer">Edit this page on GitHub</a> <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></div> <div class="last-updated"><span class="prefix">Last Updated:</span> <span class="time">9/2/2020, 8:11:30 PM</span></div></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/documentation/Ring.html" class="prev">
        Ring
      </a></span> <span class="next"><a href="/documentation/Spinner.html">
        Spinner
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.2a57d547.js" defer></script><script src="/assets/js/2.d688e986.js" defer></script><script src="/assets/js/11.03618cb3.js" defer></script>
  </body>
</html>
