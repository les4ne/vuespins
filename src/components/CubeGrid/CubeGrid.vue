<template>
  <div class="vs-cube-grid" :style="cssVars">
    <div class="vs-cube vs-cube1"></div>
    <div class="vs-cube vs-cube2"></div>
    <div class="vs-cube vs-cube3"></div>
    <div class="vs-cube vs-cube4"></div>
    <div class="vs-cube vs-cube5"></div>
    <div class="vs-cube vs-cube6"></div>
    <div class="vs-cube vs-cube7"></div>
    <div class="vs-cube vs-cube8"></div>
    <div class="vs-cube vs-cube9"></div>
  </div>
</template>

<script>
export default {
  name: 'vs-cube-grid',
  props: {
    color: {
      type: String,
      default: '#282c34'
    },
    size: {
      type: String,
      default: '50'
    },
    speed: {
      type: String,
      default: '1.3'
    }
  },
  computed: {
    cssVars() {
      return {
        '--vs-cube-grid-color': this.color,
        '--vs-cube-grid-size': this.size + 'px',
        '--vs-cube-grid-speed': this.speed + 's'
      }
    }
  }
}
</script>

<style scoped>
.vs-cube-grid {
  width: var(--vs-cube-grid-size);
  height: var(--vs-cube-grid-size);
  animation-duration: var(--vs-cube-grid-speed);
}

.vs-cube-grid .vs-cube {
  width: 33%;
  height: 33%;
  background-color: var(--vs-cube-grid-color);
  float: left;
  -webkit-animation: vs-cubeGridScaleDelay var(--vs-cube-grid-speed) infinite
    ease-in-out;
  animation: vs-cubeGridScaleDelay var(--vs-cube-grid-speed) infinite
    ease-in-out;
}
.vs-cube-grid .vs-cube1 {
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}
.vs-cube-grid .vs-cube2 {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}
.vs-cube-grid .vs-cube3 {
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
}
.vs-cube-grid .vs-cube4 {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}
.vs-cube-grid .vs-cube5 {
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}
.vs-cube-grid .vs-cube6 {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}
.vs-cube-grid .vs-cube7 {
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}
.vs-cube-grid .vs-cube8 {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}
.vs-cube-grid .vs-cube9 {
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

@-webkit-keyframes vs-cubeGridScaleDelay {
  0%,
  70%,
  100% {
    -webkit-transform: scale3D(1, 1, 1);
    transform: scale3D(1, 1, 1);
  }
  35% {
    -webkit-transform: scale3D(0, 0, 1);
    transform: scale3D(0, 0, 1);
  }
}

@keyframes vs-cubeGridScaleDelay {
  0%,
  70%,
  100% {
    -webkit-transform: scale3D(1, 1, 1);
    transform: scale3D(1, 1, 1);
  }
  35% {
    -webkit-transform: scale3D(0, 0, 1);
    transform: scale3D(0, 0, 1);
  }
}
</style>
