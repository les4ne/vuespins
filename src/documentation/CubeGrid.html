<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>CubeGrid</title>
    <meta name="generator" content="VuePress 1.5.4">
    
    <meta name="description" content="">
    <link rel="preload" href="/assets/css/0.styles.b4f6508b.css" as="style"><link rel="preload" href="/assets/js/app.d6e198f8.js" as="script"><link rel="preload" href="/assets/js/2.a0e0cdc6.js" as="script"><link rel="preload" href="/assets/js/5.2a000784.js" as="script"><link rel="prefetch" href="/assets/js/10.a3ca01ef.js"><link rel="prefetch" href="/assets/js/3.9171247f.js"><link rel="prefetch" href="/assets/js/4.918157a7.js"><link rel="prefetch" href="/assets/js/6.08c9b643.js"><link rel="prefetch" href="/assets/js/7.d5aa461e.js"><link rel="prefetch" href="/assets/js/8.f1ac1671.js"><link rel="prefetch" href="/assets/js/9.2d287674.js">
    <link rel="stylesheet" href="/assets/css/0.styles.b4f6508b.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container no-sidebar"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <!----></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><!---->  <!----> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="cubegrid"><a href="#cubegrid" class="header-anchor">#</a> CubeGrid</h1> <h2 id="basic"><a href="#basic" class="header-anchor">#</a> <strong>Basic</strong></h2> <br> <!----> <h3 id="props"><a href="#props" class="header-anchor">#</a> Props</h3> <p>Basic &quot;CubeGrid&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">-</td></tr></tbody></table> <h3 id="usage"><a href="#usage" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>CubeGrid</span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><h2 id="custom"><a href="#custom" class="header-anchor">#</a> <strong>Custom</strong></h2> <br> <!----> <h3 id="props-2"><a href="#props-2" class="header-anchor">#</a> Props</h3> <p>Custom &quot;CubeGrid&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">#fb503b</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">80</td></tr></tbody></table> <h3 id="usage-2"><a href="#usage-2" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>CubeGrid</span> <span class="token attr-name">color</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>#fb503b<span class="token punctuation">&quot;</span></span> <span class="token attr-name">size</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>80<span class="token punctuation">&quot;</span></span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> CubeGrid <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div> <footer class="page-edit"><!----> <!----></footer> <!----> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.d6e198f8.js" defer></script><script src="/assets/js/2.a0e0cdc6.js" defer></script><script src="/assets/js/5.2a000784.js" defer></script>
  </body>
</html>
