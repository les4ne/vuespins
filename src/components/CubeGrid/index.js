import CubeGrid from './CubeGrid.vue'

export default CubeGrid
