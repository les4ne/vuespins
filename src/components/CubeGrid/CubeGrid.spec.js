import { shallowMount } from '@vue/test-utils'
import CubeGrid from './CubeGrid.vue'

describe('CubeGrid.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(CubeGrid).element).toMatchSnapshot()
  })
})
