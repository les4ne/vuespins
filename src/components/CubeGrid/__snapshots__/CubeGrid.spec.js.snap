// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CubeGrid.vue renders correctly 1`] = `
<div
  class="sk-cube-grid"
  style="--vcs-cube-grid-color: #282c34; --vcs-cube-grid-size: 50px;"
>
  <div
    class="sk-cube sk-cube1"
  />
   
  <div
    class="sk-cube sk-cube2"
  />
   
  <div
    class="sk-cube sk-cube3"
  />
   
  <div
    class="sk-cube sk-cube4"
  />
   
  <div
    class="sk-cube sk-cube5"
  />
   
  <div
    class="sk-cube sk-cube6"
  />
   
  <div
    class="sk-cube sk-cube7"
  />
   
  <div
    class="sk-cube sk-cube8"
  />
   
  <div
    class="sk-cube sk-cube9"
  />
</div>
`;
